crates/matrix-sdk/src/notification_settings/rules.rs

```diff
@@ -264,16 +265,46 @@ pub(crate) mod tests {
         RoomId::parse("!AAAaAAAAAaaAAaaaaa:matrix.org").unwrap()
     }
 
-    fn build_rules(rule_list: Vec<(RuleKind, &RoomId, bool)>) -> Rules {
-        let mut rules = Rules::new(get_server_default_ruleset());
-        let mut commands = RuleCommands::new(rules.ruleset.clone());
+    fn build_ruleset(rule_list: Vec<(RuleKind, &RoomId, bool)>) -> Ruleset {
+        let mut ruleset = get_server_default_ruleset();
         for (kind, room_id, notify) in rule_list {
-            commands.insert_rule(kind, room_id, notify).unwrap();
+            let actions = if notify {
+                vec![Action::Notify, Action::SetTweak(Tweak::Sound("default".into()))]
+            } else {
+                vec![]
+            };
+            match kind {
+                RuleKind::Room => {
+                    let new_rule = NewSimplePushRule::new(room_id.to_owned(), actions);
+                    ruleset.insert(NewPushRule::Room(new_rule), None, None).unwrap();
+                }
+                RuleKind::Override => {
+                    let new_rule = NewConditionalPushRule::new(
+                        room_id.into(),
+                        vec![PushCondition::EventMatch {
+                            key: "room_id".to_owned(),
+                            pattern: room_id.to_string(),
+                        }],
+                        actions,
+                    );
+                    ruleset.insert(NewPushRule::Override(new_rule), None, None).unwrap();
+                }
+                RuleKind::Underride => {
+                    let new_rule = NewConditionalPushRule::new(
+                        room_id.into(),
+                        vec![PushCondition::EventMatch {
+                            key: "room_id".to_owned(),
+                            pattern: room_id.to_string(),
+                        }],
+                        actions,
+                    );
+                    ruleset.insert(NewPushRule::Underride(new_rule), None, None).unwrap();
+                }
+                _ => {}
+            }
```

Comment:
This is a copy/pasta from the other file, right? Can we have a test helper that does that please?

Reply:
Oh I see. I created a test helper mod for notification settings.